### PR TITLE
internal/auth: Namespace test db (fixes flake)

### PR DESCRIPTION
**Pull request by tsenart**

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
I don't see any usage of `dbtesting.GetDB` or `dbtesting.SetupGlobalTestDB` in the githuboauth package, are you sure it is required?

**Comment by tsenart** [comment]
> I don't see any usage of dbtesting.GetDB or dbtesting.SetupGlobalTestDB in the githuboauth package, are you sure it is required?

You're right, this has no effect. Perhaps we should revert instead?

**Comment by eseliger** [comment]
`gitlaboauth` needs it, I think that part we should keep?

**Comment by tsenart** [comment]
> gitlaboauth needs it, I think that part we should keep?

I don't see any references to GetDB nor SetupGlobalTestDB in that package either.

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbtesting"
 	"golang.org/x/oauth2"
 
 	"github.com/sourcegraph/sourcegraph/cmd/frontend/auth/providers"
```
Comment (unknwon) [inline enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go:13]:
```suggestion
	"golang.org/x/oauth2"

	"github.com/sourcegraph/sourcegraph/internal/database/dbtesting"
	"github.com/sourcegraph/sourcegraph/cmd/frontend/auth/providers"
```

**Review by unknwon — APPROVED** [review]

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go:13]: Can we have a linter for this?

Comment (unknwon) [reply enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go:13]: I thought [the "goimports" liniter](https://github.com/sourcegraph/sourcegraph/blob/a3f2df2bddd9bf12223af77357312c1efe688dd9/.golangci.yml#L8) should do it... but I never had the chance to figure it out.

**Review by asdine — APPROVED** [review]
Thanks!

**Comment by asdine** [comment]
> I don't see any references to GetDB nor SetupGlobalTestDB in that package either.

There is one [here](https://github.com/sourcegraph/sourcegraph/blob/aa070b3e6a17b991e3cb615e10db69343f0606ab/enterprise/cmd/frontend/internal/auth/gitlaboauth/session_test.go#L33-L33).

It's a leftover from my previous PR, we can remove it I guess. The simplest solution would be to replace it by `dbtest.NewDB(t, "")`

**Pull request merged**